src/libraries/Microsoft.Extensions.Diagnostics.Abstractions/tests/MetricsBuilderExtensionsRulesTests.cs

```diff
@@ -0,0 +1,188 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Diagnostics.Metrics;
+using Microsoft.Extensions.Options;
+using Xunit;
+
+namespace Microsoft.Extensions.Diagnostics.Tests
+{
+    public class MetricsBuilderExtensionsRulesTests
+    {
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        [InlineData("*")]
+        [InlineData("foo")]
+        public void BuilderEnableMetricsAddsRule(string meterName)
+        {
+            var services = new ServiceCollection();
+            services.AddOptions();
+            var builder = new FakeBuilder(services);
+
+            builder.EnableMetrics(meterName);
+
+            var container = services.BuildServiceProvider();
+            var options = container.GetRequiredService<IOptions<MetricsOptions>>();
+            var instance = options.Value;
+            var rule = Assert.Single(instance.Rules);
+            Assert.Equal(meterName, rule.MeterName);
+            Assert.Null(rule.InstrumentName);
+            Assert.Null(rule.ListenerName);
+            Assert.Equal(MeterScope.Local | MeterScope.Global, rule.Scopes);
+            Assert.True(rule.Enable);
+        }
+
+        [Fact]
+        public void BuilderEnableMetricsWithAllParamsAddsRule()
+        {
+            var services = new ServiceCollection();
+            services.AddOptions();
+            var builder = new FakeBuilder(services);
+
+            builder.EnableMetrics("meter", "instance", "listener", MeterScope.Local);
+
+            var container = services.BuildServiceProvider();
+            var options = container.GetRequiredService<IOptions<MetricsOptions>>();
+            var instance = options.Value;
+            var rule = Assert.Single(instance.Rules);
+            Assert.Equal("meter", rule.MeterName);
+            Assert.Equal("instance", rule.InstrumentName);
+            Assert.Equal("listener", rule.ListenerName);
+            Assert.Equal(MeterScope.Local, rule.Scopes);
+            Assert.True(rule.Enable);
+        }
+
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        [InlineData("*")]
```

Comment:
> [InlineData("*")] [](http://example.com/codeflow?start=7&length=18)

can you add `*.*`?

Reply:
@noahfalk had me remove support for `*.*`.

Reply:
Are we testing then the behavior when someone passing it as a negative case?